packages/atclient/include/atclient/monitor.h

```diff
@@ -9,6 +9,11 @@ extern "C" {
 #include <atchops/platform.h> // IWYU pragma: keep
 #include <stdbool.h>
 
+// HACK let's just get it working for now this is so wrong
```

Comment:
I was trying to avoid leaking Arduino implementation details into the main package, but I've deemed it unpractical. I've added a new macro for `ATCLIENT_SSL_TIMEOUT_EXITCODE`

Which provides:
- MBEDTLS_ERR_SSL_TIMEOUT with mbedtls
- `-101` for Arduino with bearssl (typically -1 is the only negative number used by their libraries)
- A preprocessor otherwise a preprocessor error and it must be defined externally

Reply:
We just need this number to be unique based to the other error types which may get returned by whatever implementation of tls sockets are being used

Reply:
Addressed in [40f6221](https://github.com/atsign-foundation/at_c/pull/475/commits/40f6221f743d7536dd9dd3fa102657fb0bc8230f)